Fix NumberInput borders

thyhjwb6 opened the pull request:
## Related issue
Fixes #1640 

## Overview
Fixes the `NumberInput` border.

## Reason
There are border issues:
- corners do not look sharp
- overlap from up/down buttons

## Work carried out
- [x] Refactor code
- [x] Apply fix

## Screenshot
![Screenshot 2020-11-13 at 15 59 39](https://user-images.githubusercontent.com/56078793/99092583-3e5f0180-25c9-11eb-85e2-0b594923d223.png)

thyhjwb6 committed:
refactor(UseFocus): Rename exposed fields

thyhjwb6 committed:
fix(NumberInput): Use solid border

Box shadow borders and transitions are legacy.

thyhjwb6 committed:
refactor(NumberInput): Use transient props

m7kvqbe1 reviewed (changes requested):
I think the border needs to match `TextInput`. Looks like the thickness might be slightly too low:

<img width="372" alt="Screenshot 2020-11-16 at 13 31 05" src="https://user-images.githubusercontent.com/48086589/99258163-11506000-2810-11eb-90f1-aea54f37b371.png">

Does this change also need to be applied to the up and down button focus states:

<img width="77" alt="Screenshot 2020-11-16 at 13 29 47" src="https://user-images.githubusercontent.com/48086589/99258205-1f05e580-2810-11eb-8e8f-95e0bbb0bd64.png">

thyhjwb6 committed:
refactor(StyledComponents): Reuse wrapper border

Start abstracting out reusable pieces of CSS. This
is a good example as it was inconsistent across
components.
